lms/djangoapps/courseware/tests/test_inline_analytics_integration.py

```diff
@@ -0,0 +1,424 @@
+
+        processed_data = {
+            "count_by_part": {
+                "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": {
+                    "totalIncorrectCount": 7,
+                    "totalAttemptCount": 30,
+                    "totalCorrectCount": 23,
+                },
+            },
+            "data_by_part": {
+                "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": [
+                    {
+                        "count": 7,
+                        "value_id": "choice_0",
+                        "correct": False,
+                    },
+                    {
+                        "count": 23,
+                        "value_id": "choice_1",
+                        "correct": True,
+                    },
+                ]
+            },
+            "message_by_part": {
+            },
+            "last_update_date": "Oct 15, 2014 at 10:13 UTC"
+        }
+
+        return_json = process_analytics_answer_dist(data, question_types_by_part, num_options_by_part)
+        self.assertEquals(json.loads(return_json.content), processed_data)
+
+    def test_process_analytics_answer_dist_missing_correct(self):
+
+        data = [
+            {
+                "course_id": "A/B/C",
+                "module_id": "i4x://A/B/problem/f3ed0ba7f89445ee9a83541e1fc8a2f2",
+                "part_id": "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1",
+                "correct": False,
+                "count": 7,
+                "value_id": "choice_0",
+                "answer_value_text": "Option 1",
+                "answer_value_numeric": "null",
+                "variant": None,
+                "created": "2014-10-15T101351",
+            },
+        ]
+
+        question_types_by_part = {
+            "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": "radio",
+        }
+
+        num_options_by_part = {
+            "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": 4,
+        }
+
+        processed_data = {
+            "count_by_part": {
+                "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": {
+                    "totalIncorrectCount": 7,
+                    "totalAttemptCount": 7,
+                    "totalCorrectCount": 0,
+                },
+            },
+            "data_by_part": {
+                "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": [
+                    {
+                        "count": 7,
+                        "value_id": "choice_0",
+                        "correct": False,
+                    },
+                ]
+            },
+            "message_by_part": {
+            },
+            "last_update_date": "Oct 15, 2014 at 10:13 UTC"
+        }
+
+        return_json = process_analytics_answer_dist(data, question_types_by_part, num_options_by_part)
+        self.assertEquals(json.loads(return_json.content), processed_data)
+
+    def test_process_analytics_answer_dist_variant(self):
+
+        data = [
```

Comment:
Can you please add comments to explain what type of data you're constructing & what you're testing?